tests/baselines/reference/unionOfArraysBooleanFilterCall.types

```diff
@@ -0,0 +1,95 @@
+//// [tests/cases/compiler/unionOfArraysBooleanFilterCall.ts] ////
+
+=== unionOfArraysBooleanFilterCall.ts ===
+interface Fizz {
+    id: number;
+>id : number
+
+    member: number;
+>member : number
+}
+interface Buzz {
+    id: number;
+>id : number
+
+    member: string;
+>member : string
+}
+type Falsey = "" | 0 | false | null | undefined;
+>Falsey : false | "" | 0
+>false : false
+
+
+([] as (Fizz|Falsey)[] | (Buzz|Falsey)[]).filter(Boolean); // expect type (Fizz|Buzz)[]
+>([] as (Fizz|Falsey)[] | (Buzz|Falsey)[]).filter(Boolean) : Fizz[] | Buzz[]

```

Comment:
Now returns union of arrays `Fizz[] | Buzz[]` rather than `(Fizz|Buzz)[]`.  (The comment "expect type (Fizz|Buzz)[]" is out of date).